public/app/features/dashboard-scene/scene/DashboardScene.tsx

```diff
@@ -256,13 +259,17 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
     this._changeTracker.startTrackingChanges();
   };
 
-  public saveCompleted(saveModel: Dashboard, result: SaveDashboardResponseDTO, folderUid?: string) {
-    this._initialSaveModel = {
-      ...saveModel,
-      id: result.id,
-      uid: result.uid,
-      version: result.version,
-    };
+  public saveCompleted(saveModel: Dashboard | DashboardV2Spec, result: SaveDashboardResponseDTO, folderUid?: string) {
+    if (isV2Dashboard(saveModel)) {
+      throw new Error('v2 schema not supported');
+    } else {
+      (this._serializer as DashboardSceneSerializerLike<Dashboard>).initialSaveModel = {
+        ...saveModel,
+        id: result.id,
+        uid: result.uid,
+        version: result.version,
+      };
+    }
```

Comment:
maybe this should be a method on the serializer instead?

Reply:
yup, makes sense - updated